src/System.Security.Cryptography.X509Certificates/tests/CollectionExportTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Security.Cryptography.X509Certificates.Tests
+{
+    public static class CollectionExportTests
+    {
+        private static X509Certificate2 CreateCert(string name)
+        {
+            var rsa = new RSACryptoServiceProvider(2048);
+            var csr = new CertificateRequest("CN=" + name, rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+            var cert = csr.CreateSelfSigned(DateTime.Today.AddDays(-1), DateTime.Today.AddDays(1));
+            return cert;
+        }
+
+        [Fact]
+        public static void CanAddMultipleCertsWithoutPrivate()
+        {
+            var col = new X509Certificate2Collection
+            {
+                // without private keys
+                new X509Certificate2(CreateCert("one").Export(X509ContentType.Cert)),
```

Comment:
All allocated certificates must be explicitly disposed (so that this test library can be run with finalization detection enabled to see if anything in the X509Certificates library is causing finalization)